ui_framework/src/components/button/_button.scss

```diff
@@ -0,0 +1,123 @@
+// Our base button
+.kuiButton {
+  display: inline-block;
+  appearance: none;
+  cursor: pointer;
+  height: $kuiSizeXXL;
+  text-decoration: none;
+  border: solid 1px $kuiColorPrimary;
+  border-radius: $kuiBorderRadius;
+  color: $kuiColorPrimary;
+  padding: 0 $kuiSizeS;
+  min-width: $kuiSize * 7;
+  text-align: center;
+  font-family: $kuiFontFamily;
+  transition: all $kuiAnimSpeedNormal $kuiAnimSlightBounce;
+  whitespace: nowrap;
+
+  &.kuiButton--small {
+    height: $kuiSizeXL;
+  }
+
+  .kuiButton__icon {
+    vertical-align: -2px;
+    fill: $kuiColorPrimary;
+  }
+
+  &:hover, &:focus {
+    text-decoration: underline;
+    background-color: transparentize($kuiColorPrimary, .9);
+  }
+
+  &:focus {
```

Comment:
I think the transition into and out of this focus state is cool and clearly communicates focus, but it gets weird when you click the button. It  _feels_ really uncomfortable to me because I've just pushed the button and it reacts to the push, but then looks like it got stuck. We don't need to address this in this PR, but I think later on we might need to look at an alternative to get around this behavior.